api/src/main/java/org/jboss/forge/roaster/model/util/Types.java

```diff
@@ -164,29 +176,35 @@ public static boolean areEquivalent(String left, String right)
       if (left.equals(right))
          return true;
 
-      left = stripGenerics(left);
-      right = stripGenerics(right);
-
-      String l = toSimpleName(left);
-      String r = toSimpleName(right);
+      String leftName = stripGenerics(toSimpleName(left));
+      String rightName = stripGenerics(toSimpleName(right));
 
-      String lp = getPackage(left);
-      String rp = getPackage(right);
+      String leftPackage = getPackage(left);
+      String rightPackage = getPackage(right);
 
-      if (l.equals(r))
+      if (leftName.equals(rightName))
       {
-         return !(!lp.isEmpty() && !rp.isEmpty());
+         if (!leftPackage.isEmpty() && !rightPackage.isEmpty())
+         {
+            return leftPackage.equals(rightPackage);
+         }
+         return !(!leftPackage.isEmpty() && !rightPackage.isEmpty());
       }
 
       return false;
    }
 
+   /**
+    * Calculates the simple name of the given type including all generics.
+    * 
+    * @param type the type to convert
+    * @return the simple name
+    * @throws NullPointerException if the type is {@code null}
+    * @throws IllegalArgumentException if the generic part couldn't be parsed
+    */
    public static String toSimpleName(final String type)
    {
-      if (type == null)
-      {
-         return null;
-      }
```

Comment:
This changes the method behavior

Reply:
I checked the calls of the method and I dond't see one which has a proper null handling. So I assume that null is not expected. So, we can remove this code. What do you think?

Reply:
Probably that makes sense, but it would be nice to have a test for that. I am almost sure that it does not throw NPE right now

Reply:
Yes, I'll add all necessary tests